### PR TITLE
23.lts.1+ service worker cherry-picks

### PR DESCRIPTION
**Pull request by kaidokert**
Cherry-picks series of fixes for service worker tests from trunk

b/240174245

### CODE REVIEW

**Review by sherryzy — APPROVED** [review]

File: base/single_thread_task_runner.cc [hunk 1]
```diff
@@ -47,7 +48,17 @@ void SingleThreadTaskRunner::PostBlockingTask(const base::Location& from_here,
...
-    task_finished.Wait();
+    do {
+      if (task_finished.TimedWait(base::TimeDelta::FromMilliseconds(1000))) {
+        break;
+      }
+#if !defined(COBALT_BUILD_TYPE_GOLD)
+      base::debug::StackTrace trace;
+      trace.PrintWithPrefix("[task runner deadlock]");
+#endif // !defined(COBALT_BUILD_TYPE_GOLD)
+      base::MessageLoopCurrent::ScopedNestableTaskAllower allow;
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply base/single_thread_task_runner.cc:59]: @sherryzy question - but this is a cherry-pick.

[the remark this answers is not in the document]
Comment (kaidokert) [reply base/single_thread_task_runner.cc:59]: We actually have some nightly build failures very adjacent to this, so i'm hesitant to put the full thing into an LTS branch

[the remark this answers is not in the document]
Comment (sherryzy) [reply base/single_thread_task_runner.cc:59]: Well, it's true that we should try not to use this function as much as possible. And probably not a good statement why we use it here, but we already used this function several places in service_worker_jobs.cc, and this PR is changing the other few places in that file that should call this function to fix a few bugs. It worths a follow-up 
 (b/240164388) for cleaning it up at once, but it takes much longer time(maybe we need to properly implement some event queues).

[the remark this answers is not in the document]
Comment (sherryzy) [reply base/single_thread_task_runner.cc:59]: @kaidokert Can you share with me the nightly build that fails probably related to this? Are they related to service worker WPT?

[the remark this answers is not in the document]
Comment (kaidokert) [reply base/single_thread_task_runner.cc]: Shared in bugtracker. This PR is removed from the cherrypick.

**Pull request merged**